silo-core/contracts/incentives/base/BaseIncentivesController.sol

```diff
@@ -0,0 +1,299 @@
+    function claimRewards(address _to, bytes32[] calldata _programIds)
+        external
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        if (_to == address(0)) revert InvalidToAddress();
+
+        accruedRewards = _accrueRewardsForPrograms(_to, _programIds);
+        _claimRewards(msg.sender, msg.sender, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewards(address _to, string[] calldata _programNames)
+        external
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        if (_to == address(0)) revert InvalidToAddress();
+
+        bytes32[] memory programIds = _getProgramsIds(_programNames);
+        accruedRewards = _accrueRewardsForPrograms(_to, programIds);
+        _claimRewards(msg.sender, msg.sender, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewardsOnBehalf(address _user, address _to)
+        external
+        onlyAuthorizedClaimers(msg.sender, _user)
+        inputsValidation(_user, _to)
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        accruedRewards = _accrueRewards(_user);
+        _claimRewards(msg.sender, _user, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewardsOnBehalf(address _user, address _to, bytes32[] calldata _programIds)
+        external
+        onlyAuthorizedClaimers(msg.sender, _user)
+        inputsValidation(_user, _to)
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        accruedRewards = _accrueRewardsForPrograms(_user, _programIds);
+        _claimRewards(msg.sender, _user, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewardsOnBehalf(address _user, address _to, string[] calldata _programNames)
+        external
+        onlyAuthorizedClaimers(msg.sender, _user)
+        inputsValidation(_user, _to)
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        bytes32[] memory programIds = _getProgramsIds(_programNames);
+        accruedRewards = _accrueRewardsForPrograms(_user, programIds);
+        _claimRewards(msg.sender, _user, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewardsToSelf() external returns (AccruedRewards[] memory accruedRewards) {
+        accruedRewards = _accrueRewards(msg.sender);
+        _claimRewards(msg.sender, msg.sender, msg.sender, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewardsToSelf(bytes32[] calldata _programIds)
+        external
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        accruedRewards = _accrueRewardsForPrograms(msg.sender, _programIds);
+        _claimRewards(msg.sender, msg.sender, msg.sender, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewardsToSelf(string[] calldata _programNames)
```

Comment:
We should choose either bytes32 or string as input and reduce the interface.

Reply:
Done. Methods that worked with bytes32 removed.